app/models/post.rb

```diff
@@ -1,5 +1,15 @@
 class Post < ApplicationRecord
-  belong_to :users
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
   has_many :comments
   has_many :likes
+
+  after_save :update_post_counter
+
+  def update_post_counter
+    User.find(author_id).increment!(:posts_counter)
+  end
```

Comment:
- [ ] [OPTIONAL] I will like to recommend that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private .

```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```

- You can also explore the same for the `Comment` and `Like` models respectively.